### PR TITLE
Added deployment strategy for Linux Premium Plan

### PR DESCRIPTION
**Pull request by eaarora-ms**
Reference : https://github.com/microsoft/azure-pipelines-tasks/pull/11257
Fixes : https://github.com/microsoft/azure-pipelines-tasks/issues/11256

### CODE REVIEW

**Comment by eaarora-ms** [comment]
@SumiranAgg  Tested the changes from Azure Function App. The package is getting deployed successfully.

File: Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ var zipUtility = require('azurermdeploycommon/webdeployment-common/ziputility.js
 
 const linuxFunctionStorageSetting: string = '-WEBSITES_ENABLE_APP_SERVICE_STORAGE true';
 const linuxFunctionRuntimeSettingName: string = '-FUNCTIONS_WORKER_RUNTIME ';
+const premiumPlanRunsFromPackage: string = ' -WEBSITE_RUN_FROM_PACKAGE true';
```
Comment (niadak) [inline Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts]: Is this relevant for other type of Apps or this is specific to FunctionApp?

Comment (eaarora-ms) [reply Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts]: This is specific to function app

File: Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts [hunk 2]
```diff
@@ -32,6 +33,10 @@ export class BuiltInLinuxWebAppDeploymentProvider extends AzureRmWebAppDeploymen
             linuxFunctionRuntimeSetting = linuxFunctionRuntimeSettingName + linuxFunctionRuntimeSettingValue.get(this.taskParams.RuntimeStack);
         }
         var linuxFunctionAppSetting = linuxFunctionRuntimeSetting + linuxFunctionStorageSetting;
+        if(this.taskParams.isPremium) {
```
Comment (SumiranAgg) [inline Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts]: add !! before variable in if

File: Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts (hunk 1 above)
Comment (SumiranAgg) [inline Tasks/AzureFunctionAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts]: should value be 1 or true

File: Tasks/AzureFunctionAppV1/taskparameters.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { AzureRMEndpoint } from 'azurermdeploycommon/azure-arm-rest/azure-arm-en
 import { AzureResourceFilterUtility } from 'azurermdeploycommon/operations/AzureResourceFilterUtility';
 import { AzureAppService } from 'azurermdeploycommon/azure-arm-rest/azure-arm-app-service';
 const skuDynamicValue: string = 'dynamic';
+const skuElasticPremiumValue: string = 'elasticpremium';
```
Comment (SumiranAgg) [inline Tasks/AzureFunctionAppV1/taskparameters.ts]: Is there anything called just elastic?

**Pull request merged**